index.js

```diff
@@ -73,13 +55,28 @@ Sibyl.prototype.open = function (address, cb) {
   assert.equal(typeof cb, 'function', 'sibyl.open: cb should be type function')
 
   this._initialize(address, (err, protocol, location, version) => {
-    if (err) cb(err)
-    else {
-      this._fetch(protocol, location, version, (err) => {
-        if (err) cb(err)
-        else this._finalize(cb)
+    if (err) return cb(err)
+    var directory = this.directory
+    var name = this.name
+    fetch(protocol, location, version, directory, (err) => {
+      if (err) return cb(err)
+      compile(directory, function (err) {
+        if (err) return cb(err)
+        build(directory, name, function (err) {
+          if (err) return cb(err)
+          run(name, function (err, data) {
+            if (err) return cb(err)
+            open(`http://localhost:${data.port}`)
```

Comment:
using `async.series` would be a great match here. Part of https://github.com/yoshuawuyts/async-collection